src/thunderbots/software/ai/world/game_state.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <geom/point.h>
+#include <util/refbox_constants.h>
+
+/**
+ * @brief Holds the state of the game according to the referee
+ *
+ * @details Contains information on what period of the game it is, what type of
+ * play to run, team scores, time remaining, etc.  During normal gameplay, the
+ * information in this class is received from the [ssl
+ * refbox](https://github.com/RoboCup-SSL/ssl-refbox)
+ * program over the network.
+ *
+ * Taken from
+ */
+class GameState
+{
+   public:
+    enum State
+    {
+        HALT,    // Robots must not move
+        STOP,    // Robots must stay 500mm away from ball
+        SETUP,   // Robots not on starting team must stay 500mm away from ball
+        READY,   // A robot on the starting team may kick the ball
+        PLAYING  // Normal play
+    };
+
+    // Types of restarts
+    enum Restart
+    {
+        NONE,
+        KICKOFF,
+        DIRECT,
+        INDIRECT,
+        PENALTY,
+        BALL_PLACEMENT
+    };
+
+    State state;
+    Restart restart;
+
+    // True if our team can kick the ball during a restart
+    bool ourRestart;
+
+    Point ballPlacementPoint;
+
+    GameState()
+    {
+        state            = HALT;
+        restart          = NONE;
+        ourRestart       = false;
+    }
+
+    void updateRefboxGameState(RefboxGameState gameState);
+
+    // Rule queries
+
+    bool halt() const;
+
+    bool stopped() const;
+
+    bool playing() const;
+
+    bool kickoff() const;
+
+    bool penalty() const;
+
+    bool ballPlacement() const;
+
+    bool isOurRestart() const;
```

Comment:
the `is` at the beginning of `isOurRestart` seems inconsistent with the rest of these function names (which don't have `is`). Can we do without it?

Reply:
Agree with @garethellis0

Reply:
renamed as per @garethellis0 's suggestion